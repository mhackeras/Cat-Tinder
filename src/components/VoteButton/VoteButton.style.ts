import {StyleSheet} from 'react-native';
import {colors} from '../../res/colors';

export default StyleSheet.create({
  container: {
    width: 54,
    height: 54,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 100,
    backgroundColor: colors.white,
    shadowColor: colors.shadow,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 1,
    elevation: 16,
  },
  image: {
    width: 32,
    height: 32,
  },
});
