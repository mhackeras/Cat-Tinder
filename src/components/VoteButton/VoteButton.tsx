import React from 'react';
import {Image, TouchableOpacity, ViewStyle} from 'react-native';
import styles from './VoteButton.style';

interface Props {
  variant: 'heart' | 'x';
  onPress: () => void;
  customStyle?: ViewStyle;
}

const VoteButton = ({variant, onPress, customStyle}: Props) => {
  const icon =
    variant === 'heart'
      ? require('../../res/pngs/vote/heart.png')
      : require('../../res/pngs/vote/x.png');
  const finalStyle = Object.assign({}, styles.container, customStyle);

  return (
    <TouchableOpacity style={finalStyle} onPress={onPress}>
      <Image source={icon} style={styles.image} />
    </TouchableOpacity>
  );
};

export default VoteButton;
