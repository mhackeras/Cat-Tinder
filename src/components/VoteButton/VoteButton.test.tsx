import React from 'react';
import {Image} from 'react-native';
import {act, render, fireEvent} from '@testing-library/react-native';

import VoteButton from './VoteButton';

describe('VoteButton', () => {
  it('renders a vote button with heart icon', () => {
    const callback = jest.fn();

    const tree = render(<VoteButton variant={'heart'} onPress={callback} />);
    expect(tree).toMatchSnapshot();
  });

  it('fires click event for a hear button', async () => {
    const callback = jest.fn();

    const {getByTestId} = render(
      <VoteButton variant="heart" onPress={callback} />,
    );

    expect(callback).toBeCalledTimes(0);
    await act(() => fireEvent.press(getByTestId('vote-button')));
    expect(callback).toBeCalledTimes(1);
  });
});
