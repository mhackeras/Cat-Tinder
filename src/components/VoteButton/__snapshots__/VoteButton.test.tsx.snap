// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteButton renders a vote button with heart icon 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 100,
      "elevation": 16,
      "height": 54,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "rgba(191, 191, 192, 0.3)",
      "shadowOffset": Object {
        "height": 10,
        "width": 0,
      },
      "shadowOpacity": 1,
      "width": 54,
    }
  }
  testID="vote-button"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/res/pngs/vote/heart.png",
      }
    }
    style={
      Object {
        "height": 32,
        "width": 32,
      }
    }
  />
</View>
`;
